[DOCS] Add search and filtering tutorial/quickstart, edit filtering page

leemthompo opened the pull request:
## Ꙫ [URL PREVIEW](https://elasticsearch_bk_114353.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/full-text-filter-tutorial.html)


- First draft was very WIPpy
- Based on SME feedback https://github.com/elastic/elasticsearch/pull/114353/commits/92e47af3da87e2e4948863f57677789eb3e2176b represents significant tightening and focus of tutorial
- Includes minor update to the filtering overview 🚗 


## Todo

- [x] Change filename, ID

leemthompo committed:
Add search and filtering tutorial, edit filtering page

leemthompo committed:
Fix example syntax

leemthompo committed:
Fix syntax

leemthompo commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'main' into qs-full-text

kderusso commented on docs/reference/query-dsl/query_filter_context.asciidoc:
```diff
@@ -29,26 +29,35 @@ parameter, such as the `query` parameter in the
 
-*  __Does this +timestamp+ fall into the range 2015 to 2016?__
-*  __Is the +status+  field set to ++"published"++__?
+A filter answers the binary question “Does this document match this query clause?”. The anwser is simply "yes" or "no".
```
```suggestion
A filter answers the binary question “Does this document match this query clause?”. The answer is simply "yes" or "no".
```

kderusso commented on docs/reference/query-dsl/query_filter_context.asciidoc:
```diff
@@ -29,26 +29,35 @@ parameter, such as the `query` parameter in the
+Common filter applications include:
+
+* Date range checks: for example is the `timestamp` field between 2015 and 2016
+* Status checks: for example is the `status` field equals "published"
```
```suggestion
* Status checks: for example is the `status` field equal to "published"
```

kderusso commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+[TIP]
+====
+Full-text search is powered by <<analysis,text analysis>>. 
+Text analysis normalizes and standardizes text data so it can be efficiently stored in an inverted index and searched in near real-time.
```
Link to docs defining inverted index here?

leemthompo replied:
We don't have this today :sad_panda:.

I would love to have a page about data structures in the [basics section](https://www.elastic.co/guide/en/elasticsearch/reference/current/elasticsearch-intro.html) that summarizes how Elasticsearch stores different data types in optimized structures: inverted index for text, BKD trees for geospatial data, HNSW for vectors etc.

A lot of this core stuff is buried in blogs.

Another disappointing learning doing some of this work is that at some point the Definitive Guide was deprecated but not all of the essential basics were ported over into the docs! 

See https://github.com/elastic/elasticsearch-definitive-guide/blob/master/052_Mapping_Analysis/35_Inverted_index.asciidoc. Need to restore this to the docs ASAP in the right spot.

kderusso commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+
+[source,console]
+----
+POST /cooking_blog/_bulk
```
Would this make more sense in a notebook?

leemthompo replied:
You mean the whole tutorial? For sure, the plan is to have notebook versions too once have a few of these :)

kderusso commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+[[full-text-tutorial-multi-match]]
+=== Search across multiple fields at once
+
+A <<query-dsl-multi-match-query,`multi_match`>> query runs the same query against multiple fields.
```
Worth adding field boosts here. I would personally recommend most text search users use `multi_match` over `match`.

leemthompo replied:
>  I would personally recommend most text search users use multi_match over match.

Cool, what would be the concise explanation for this recommendation?

leemthompo replied:
Handled in https://github.com/elastic/elasticsearch/pull/114353/commits/92e47af3da87e2e4948863f57677789eb3e2176b

kderusso commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+[[full-text-tutorial-bool-query]]
+=== Combine multiple search criteria
+
+A <<query-dsl-bool-query,`bool`>> query combines multiple queries using boolean logic.
```
It would be nice to have a final bool query at the end that combines a bunch of stuff - multi match, fuzziness, filters, etc.

leemthompo replied:
Added in https://github.com/elastic/elasticsearch/pull/114353/commits/92e47af3da87e2e4948863f57677789eb3e2176b

kderusso reviewed (commented):
Nice start so far!

leemthompo committed:
Merge branch 'main' into qs-full-text

leemthompo committed:
Typos

Co-authored-by: Kathleen DeRusso <kathleen.derusso@elastic.co>

carlosdelest commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+[discrete]
+[[full-text-tutorial-match-query]]
+=== Perform basic full-text searches
+
```
```suggestion

Full text search consists of performing text queries on different document fields.
These queries will score each document according to how the query text matches the document content.
Different queries provide different ways of looking for the text information and scoring the document.

```

carlosdelest commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+==== `match` query
+
+The <<query-dsl-match-query, `match`>> query is the standard query for full-text AKA "lexical" search.
+Input is analyzed before performing the search by tokenizing text, removing possessives, converting to lowercase, filtering out stop words, and applying stemming.
```
```suggestion
The query text will be analyzed according to the analyzer configuration specified on each field (or at query time). This provides an opportunity for filtering out stop words, applying stemming to look for different word variations, using synonyms, remove uppercase and lowercase distinctions, etc.
```

carlosdelest commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+    "match": {
+      "description": {
+        "query": "fluffy pancakes", <1>
+        "analyzer": "standard" <2>
```
Should we move analyzers to a different section? I think that could be initially confusing. Adding analyzers as a secondary step for fine-tuning relevance, applying synonyms, etc could be interesting to have once full-text search is understood.

leemthompo replied:
Indeed it might even be out of scope of this 101 quick start to dig into analyzers.

leemthompo replied:
Just _knowing_ that analyzers are a thing after working thru these examples might be good enough for this section.

carlosdelest commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+}
+----
+// TEST[continued]
+
```
Maybe add some links to other full text queries, like match phrase, query string, intervals?

carlosdelest commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+
+<<filter-context,Filtering>> allows you to narrow down your search results based on exact criteria.
+Unlike full-text searches, filters are binary (yes/no) and do not affect the relevance score.
+Filters execute faster than queries.
```
```suggestion
Filters execute faster than queries as no scoring needs to be calculated for excluded results.
```

carlosdelest commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+}
+----
+// TEST[continued]
+<1> The `term` query has zero flexibility. For example, here the queries `maria` or `maria rodriguez` would have zero hits.
```
I'd say this depends on the analyzer used. That would be a good entry point for an analyzers section.

carlosdelest commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+[discrete]
+=== Perform basic aggregations
+
+Aggregations provide summary statistics and analytics on your data.
```
```suggestion
Aggregations provide summary statistics and analytics on your search results.
```

leemthompo commented on docs/reference/query-dsl/query_filter_context.asciidoc:
```diff
@@ -29,26 +29,35 @@ parameter, such as the `query` parameter in the
+
+* `filter` or `must_not` parameters in <<query-dsl-bool-query,`bool`>>  queries
+* `filter` parameter in <<query-dsl-constant-score-query,`constant_score`>> queries
+* <<search-aggregations-bucket-filter-aggregation,`filter`>> aggregations
```
It might be a bit too much in this 101 searching and filtering intro! This is the first tentative toe in the search paddling pool. 🦶 

Currently I'm thinking the next basics "quickstart"  could be a`semantic_text` quickstart  (maybe ELSER + hybrid via rrf). That would be a toy example, API-only, with no reindexing or anything, just the happiest path — and then link to **Search your data** for the full-blown semantic search examples.

leemthompo replied:
I think there might be a little too much in this first draft already, let's see how we fell once I've incorporated the first round of feedback.

leemthompo replied:
@carlosdelest sorry I forgot that I edited this page too, so I glanced over this comment as if it related to the tutorial!

Indeed we should add whatever else is missing from this query_filter_context.asciidoc page too 😄

leemthompo commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+[[full-text-tutorial-prefix-query]]
+=== Search by prefixes
+
+A <<query-dsl-prefix-query,`prefix`>> query enables you to quickly narrow down results to documents that begin with a specified prefix.
```
💯 agree that there's a bit too much in here already.

leemthompo commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+----
+GET /cooking_blog/_search
+{
+  "size": 0,
```
```suggestion
  "size": 0, <1>
```

leemthompo commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+}
+----
+// TEST[continued]
+
```
```suggestion
<1> Here you just want the aggregated information on search results, and don't need to return the actual search results themselves.

```

leemthompo committed:
Apply Carlos' suggestions

Co-authored-by: Carlos Delgado <6339205+carlosdelest@users.noreply.github.com>

leemthompo committed:
Fix syntax

Mikep86 commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,435 @@
+  "query": {
+    "bool": {
+      "filter": [
+        { "term": { "category.keyword": "Breakfast" } }
```
Could be good to call out the usage of the `keyword` multi-field here

Mikep86 commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,435 @@
+// TEST[continued]
+<1> Because the `title` field is analyzed at index time, it's been processed by the `lowercase` filter. The prefix query isn't analyzed and would return zero hits if you passed "Spicy" here, without setting `case_insensitive` to `true`.
+
+[discrete]
+[[full-text-tutorial-query-string]]
+=== Perform advanced searches with query strings
+
+A <<query-dsl-query-string-query,`query_string`>> query supports complex queries with a compact syntax.
+Use `query_string` for advanced, multi-field searches with complex syntax.
+
+[source,console]
+----
+GET /cooking_blog/_search
+{
+  "query": {
+    "query_string": {
+      "fields": ["title", "description", "tags"], <1>
+      "query": "(vegetarian OR vegan) AND (curry OR *fry) AND rating:>=4.5" <2>
+    }
+  }
+}
+----
+// TEST[continued]
+<1> Searches across multiple fields simultaneously
+<2> Combines boolean logic, wildcards, and `range` queries:
+   - `(vegetarian OR vegan)`: Matches either dietary preference
+   - `(curry OR *fry)`: Uses wildcard for various cooking methods
+   - `rating:>=4.5`: Filters for highly-rated recipes
```
There's a lot in this example, we should expand the description

leemthompo replied:
might be one for more advanced follow-up

Mikep86 commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,435 @@
+[discrete]
+=== Perform basic aggregations
+
+Aggregations provide summary statistics and analytics on your search results.
```
Can we link to a page with more detailed info on aggregations here?

leemthompo replied:
out of scope now, we'll have an aggregations heavy tutorial methinks

Mikep86 reviewed (commented):
Great start, thank you for doing this! I think the tutorial could benefit from a tighter "story arc", so to speak. Since this is a targeted at a brand new user, I say we try to cover less material with more detail, including putting the search responses in the documentation so that the user can see matching docs and we can explain why they match. I think it could also help to create a business scenario for each section to ground the tutorial in the real world.

We can stretch this material across multiple pages too. For example, there's no need to cover aggregations in the "searching and filtering" page, we can make an aggregation-specific tutorial page instead.

leemthompo commented on docs/reference/quickstart/full-text-tutorial.asciidoc:
```diff
@@ -0,0 +1,428 @@
+----
+GET /cooking_blog/_search
+{
+  "size": 0,
```
Out of scope now that we've decided to trim aggregations from this tutorial, but relevant for follow-up  aggregations tutorial

leemthompo committed:
Sharpen tutorial focus, explain queries and responses, etc

Added detailed examples of how to configure the cooking_blog index with text and keyword fields for different properties, improving exact matching and filtering capabilities.
Expanded explanations and added concrete examples and responses to illustrate how match and multi_match queries work.
Provided detailed examples on how to use filter in queries, including exact matches for categories and date ranges.
Added a detailed example of a complex bool query combining multiple search criteria, tailored to the cooking blog scenario.
Added learn more section

leemthompo commented:
Tried to generally sharpen this tutorial, make it more user-focused, removing non-essential steps and going into more detail, with more annotated responses in https://github.com/elastic/elasticsearch/pull/114353/commits/92e47af3da87e2e4948863f57677789eb3e2176b

leemthompo committed:
Fix test?

leemthompo committed:
docscheck CI whackamole

leemthompo committed:
Add links to Query DSL, mapping types, and bool query; expanded multi-fields explanation.

leemthompo committed:
Add link to search api

leemthompo committed:
Will docscheck work now?

leemthompo committed:
Use substitutions in response test

leemthompo committed:
Comment out flakey test

leemthompo committed:
Add step numbers

kderusso commented on docs/reference/query-dsl/query_filter_context.asciidoc:
```diff
@@ -29,26 +29,35 @@ parameter, such as the `query` parameter in the
+Common filter applications include:
+
+* Date range checks: for example is the `timestamp` field between 2015 and 2016
+* Status checks: for example is the `status` field equal to "published"
```
I wonder if people who aren't really familiar with the concept of filters will equate "status checks" with other field data, such as authors' name. We could consider renaming "Status checks" to something else like "Structured data checks" or "Specific field value checks" or something along those lines? WDYT?

leemthompo replied:
sounds good

kderusso commented on docs/reference/query-dsl/query_filter_context.asciidoc:
```diff
@@ -29,26 +29,35 @@ parameter, such as the `query` parameter in the
-<<query-dsl-bool-query,`bool`>> query, the `filter` parameter in the
-<<query-dsl-constant-score-query,`constant_score`>> query, or the
-<<search-aggregations-bucket-filter-aggregation,`filter`>> aggregation.
+Filters are particularly effective for querying structured data and implementing "must have" criteria in complex searches.
```
It could be useful to define here what we mean by "structured data."

leemthompo replied:
added!

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+<titleabbrev>Basics: Full-text search and filtering</titleabbrev>
+++++
+
+This is a hands-on introduction to the basics of full-text search with {es}, also known as _lexical search_, using the <<search-search,`_search` API>> and <<query-dsl,Query DSL>>.
```
Philosophical question here, but should we refactor new tutorials to use retrievers now that they will be GA soon?

leemthompo replied:
Maybe future tutorials, honestly I'd like an aligned answer from product because the query languages decision for new users is getting complex between query DSL / retrievers / ES|QL

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+    "properties": {
+      "title": {
+        "type": "text",
+        "analyzer": "standard", <1>
```
😌 I was very happy to see that the copy button copies it in pastable format for dev console even with annotations 😅

leemthompo replied:
Yes this is a QOL thing that somebody wise solved a long time ago in a galaxy far far away!

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+<1> The `standard` analyzer is used by default for `text` fields if an `analyzer` isn't specified. It's included here for demonstration purposes.
+<2> <<multi-fields,Multi-fields>> are used to index `text` fields as both `text` and `keyword` <<mapping-types,data types>>. This enables both full-text search and exact matching/filtering on the same field.
+Note that if you used <<dynamic-field-mapping,dynamic mapping>>, these multi-fields would be created automatically.
+<3> The `ignore_above` parameter prevents indexing values longer than 256 characters in the `keyword` field. Again this is the default value, but it's included here for for demonstration purposes.
```
Is this line correct? I think of `mapping explosion` as too high of a total number of mapped fields, not the size of a particular field. Looking at the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/ignore-above.html) it appears to confirm that its purpose is to save space and also avoid potentially running into issues with Lucene's term-byte limit.

leemthompo replied:
nice thanks for clarifying, updating to add link to doc and correct misstatement

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+{"index":{"_id":"3"}}
+{"title":"Classic Beef Stroganoff: A Creamy Comfort Food","description":"Indulge in this rich and creamy beef stroganoff. Tender strips of beef in a savory mushroom sauce, served over a bed of egg noodles. It's the ultimate comfort food for chilly evenings.","author":"Emma Watson","date":"2023-05-10","category":"Main Course","tags":["beef","pasta","comfort food"],"rating":4.7}
+{"index":{"_id":"4"}}
+{"title":"Vegan Chocolate Avocado Mousse","description":"Discover the magic of avocado in this rich, vegan chocolate mousse. Creamy, indulgent, and secretly healthy, it's the perfect guilt-free dessert for chocolate lovers.","author":"Alex Green","date":"2023-05-15","category":"Dessert","tags":["vegan","chocolate","avocado","healthy dessert"],"rating":4.5}
```
Oooh, I see that this could be a good setup for `vegetarian` mapping to `vegan` in future iterations to improve recall 😉

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+  }
+}
+----
+// TEST[skip:Flakey test]
```
Why is this test flakey?

leemthompo replied:
rerunning and will check

leemthompo replied:
<details>
  <summary>Expand to see build fail</summary>



```	FAILURE: Build failed with an exception.
2024-10-23 13:35:56 CEST	
2024-10-23 13:35:56 CEST	* What went wrong:
2024-10-23 13:35:56 CEST	Execution failed for task ':docs:yamlRestTest'.
2024-10-23 13:35:56 CEST	> There were failing tests. See the report at: file:///dev/shm/bk/bk-agent-prod-gcp-1729682825607451781/elastic/elasticsearch-pull-request/docs/build/reports/tests/yamlRestTest/index.html
2024-10-23 13:35:56 CEST	
2024-10-23 13:35:56 CEST	* Exception is:
2024-10-23 13:35:56 CEST	org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':docs:yamlRestTest'.
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:130)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:293)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:128)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:116)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:459)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:376)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
2024-10-23 13:35:56 CEST	        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
2024-10-23 13:35:56 CEST	        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
2024-10-23 13:35:56 CEST	        at java.base/java.lang.Thread.run(Thread.java:1583)
2024-10-23 13:35:56 CEST	Caused by: org.gradle.api.internal.exceptions.MarkedVerificationException: There were failing tests. See the report at: file:///dev/shm/bk/bk-agent-prod-gcp-1729682825607451781/elastic/elasticsearch-pull-request/docs/build/reports/tests/yamlRestTest/index.html
2024-10-23 13:35:56 CEST	        at org.gradle.api.tasks.testing.AbstractTestTask.handleTestFailures(AbstractTestTask.java:706)
2024-10-23 13:35:56 CEST	        at org.gradle.api.tasks.testing.AbstractTestTask.handleCollectedResults(AbstractTestTask.java:540)
2024-10-23 13:35:56 CEST	        at org.gradle.api.tasks.testing.AbstractTestTask.executeTests(AbstractTestTask.java:535)
2024-10-23 13:35:56 CEST	        at org.gradle.api.tasks.testing.Test.executeTests(Test.java:731)
2024-10-23 13:35:56 CEST	        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
2024-10-23 13:35:56 CEST	        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:125)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.TaskExecution$3.run(TaskExecution.java:244)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.TaskExecution.executeAction(TaskExecution.java:229)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.TaskExecution.executeActions(TaskExecution.java:212)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.TaskExecution.executeWithPreviousOutputFiles(TaskExecution.java:195)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.TaskExecution.execute(TaskExecution.java:162)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:105)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:44)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:56)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:56)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:44)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:42)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:75)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:50)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.PreCreateOutputParentsStep.execute(PreCreateOutputParentsStep.java:28)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:67)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:37)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:61)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:26)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:69)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.CaptureOutputsAfterExecutionStep.execute(CaptureOutputsAfterExecutionStep.java:46)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:40)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:29)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:189)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BuildCacheStep.executeAndStoreInCache(BuildCacheStep.java:145)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BuildCacheStep.lambda$executeWithCache$4(BuildCacheStep.java:101)
2024-10-23 13:35:56 CEST	        at java.base/java.util.Optional.orElseGet(Optional.java:364)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BuildCacheStep.lambda$executeWithCache$5(BuildCacheStep.java:101)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.Try$Success.map(Try.java:175)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BuildCacheStep.executeWithCache(BuildCacheStep.java:85)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BuildCacheStep.lambda$execute$0(BuildCacheStep.java:74)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.Either$Left.fold(Either.java:115)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.caching.CachingState.fold(CachingState.java:62)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:73)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:48)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:46)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:35)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:75)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$2(SkipUpToDateStep.java:53)
2024-10-23 13:35:56 CEST	        at java.base/java.util.Optional.orElseGet(Optional.java:364)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:53)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:35)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:49)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ResolveIncrementalCachingStateStep.executeDelegate(ResolveIncrementalCachingStateStep.java:27)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:71)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AbstractResolveCachingStateStep.execute(AbstractResolveCachingStateStep.java:39)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:65)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:36)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:107)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:56)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:64)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AbstractCaptureStateBeforeExecutionStep.execute(AbstractCaptureStateBeforeExecutionStep.java:43)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.executeWithNonEmptySources(AbstractSkipEmptyWorkStep.java:125)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:61)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AbstractSkipEmptyWorkStep.execute(AbstractSkipEmptyWorkStep.java:36)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:36)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.LoadPreviousExecutionStateStep.execute(LoadPreviousExecutionStateStep.java:23)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:75)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.HandleStaleOutputsStep.execute(HandleStaleOutputsStep.java:41)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.lambda$execute$0(AssignMutableWorkspaceStep.java:35)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.TaskExecution$4.withWorkspace(TaskExecution.java:289)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:31)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.AssignMutableWorkspaceStep.execute(AssignMutableWorkspaceStep.java:22)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:40)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ChoosePipelineStep.execute(ChoosePipelineStep.java:23)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.lambda$execute$2(ExecuteWorkBuildOperationFiringStep.java:67)
2024-10-23 13:35:56 CEST	        at java.base/java.util.Optional.orElseGet(Optional.java:364)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:67)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.ExecuteWorkBuildOperationFiringStep.execute(ExecuteWorkBuildOperationFiringStep.java:39)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:46)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:34)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:48)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:35)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:61)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:127)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:116)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:209)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:166)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
2024-10-23 13:35:56 CEST	        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:331)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:318)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.lambda$execute$0(DefaultTaskExecutionGraph.java:314)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:85)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:314)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:459)
2024-10-23 13:35:56 CEST	        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:376)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
2024-10-23 13:35:56 CEST	        at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:48)
2024-10-23 13:35:56 CEST	        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
2024-10-23 13:35:56 CEST	        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
2024-10-23 13:35:56 CEST	        at java.base/java.lang.Thread.run(Thread.java:1583)
2024-10-23 13:35:56 CEST```

</details>

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+    ]
+----
+// TEST[continued]
+<1> The `hits` object contains the total number of matching documents and their relation to the total. <1> The `relation` field can have one of three values: "eq", "gte", or "lte".
```
```suggestion
<1> The `hits` object contains the total number of matching documents and their relation to the total. The `relation` field can have one of three values: "eq", "gte", or "lte":
```

Some rendering weirdness here: 
<img width="860" alt="image" src="https://github.com/user-attachments/assets/1837aa98-3889-4391-868a-37866c3958fd">

Also a nitpick for `.` vs `:`.

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+----
+// TEST[continued]
+<1> The `hits` object contains the total number of matching documents and their relation to the total. <1> The `relation` field can have one of three values: "eq", "gte", or "lte".
+* "eq": The `value` is exactly equal to the total number of hits.
```
Do we want "eq" or `eq`?

leemthompo replied:
the latter!

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+// TEST[continued]
+<1> The `hits` object contains the total number of matching documents and their relation to the total. <1> The `relation` field can have one of three values: "eq", "gte", or "lte".
+* "eq": The `value` is exactly equal to the total number of hits.
+* "gte": The `value` is greater than or equal to the total number of hits. This occurs when the number of hits exceeds the configured max result window.
```
This may be a little misleading. First of all, users might not know what the configured max result window is, second, this can be different depending on properties such as `track_total_hits`. 

I think the summary in the [track_total_hits](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-your-data.html#track-total-hits) doc is more correct: 

> When set to true the search response will always track the number of hits that match the query accurately (e.g. total.relation will always be equal to "eq" when track_total_hits is set to true). Otherwise the "total.relation" returned in the "total" object in the search response determines how the "total.value" should be interpreted. A value of "gte" means that the "total.value" is a lower bound of the total hits that match the query and a value of "eq" indicates that "total.value" is the accurate count.

I wonder if this is a case where minimal explanation is better?

leemthompo replied:
> I wonder if this is a case where minimal explanation is better?

Definitely wandered deep into overkill territory here I think

leemthompo replied:
I vote for removing the list and explanations of `relation` and linking to track total hits for those interested 👍

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+  "query": {
+    "multi_match": {
+      "query": "vegetarian curry",
+      "fields": ["title^3", "description^2", "tags"] <1>
```
I wonder if we should do a "vanilla" `multi_match` first that just lists the fields with no boosts, then explain that the title is a little more important and then show the boosts? WDYT?

leemthompo replied:
totally

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+}
+----
+// TEST[continued]
+<1> Note the use of `category.keyword` here. This refers to the `keyword` multi-field of the `category` field, ensuring an exact, case-sensitive match.
```
Link to document on keyword mapped field type?

leemthompo replied:
yes!

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+
+[TIP]
+====
+Using the `.keyword` suffix accesses the unanalyzed version of a field.
```
This will work in dynamically mapped fields and your example, but not if a multifield is not mapped.

leemthompo replied:
tried to make note hyperclear

kderusso commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,588 @@
+}
+----
+// TEST[continued]
+<1> The `term` query has zero flexibility. For example, here the queries `maria` or `maria rodriguez` would have zero hits.
```
```suggestion
<1> The `term` query has zero flexibility. For example, here the queries `maria` or `maria rodriguez` (due to case sensitivity) would have zero hits.
```

kderusso reviewed (commented):
Looks great, nice progress!

leemthompo commented:
@kderusso random question: when or where in a new user's journey would it be helpful to introduce the [explain API](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-explain.html)?

kderusso commented:
> @kderusso random question: when or where in a new user's journey would it be helpful to introduce the [explain API](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-explain.html)?

@leemthompo TBH, I could see the `explain` API deserving its own page with an explanation of how to use it?

leemthompo commented:
> TBH, I could see the explain API deserving its own page with an explanation of how to use it?

Ooooh nice and meta let's explain the explain 😃

leemthompo committed:
Clarify filtering overview
* Expanded explanation of structured data in Elasticsearch:
 - Added examples of data types for filtering
 - Enhanced clarity around structured data concepts

* Enhanced "Common filter applications" section:
 - Added specific field value checking example
 - More comprehensive coverage of filtering use cases

leemthompo committed:
Address Kathleen's feedback

* Updates to full-text-filtering-tutorial:
 - Added ignore_above parameter documentation link
 - Clarified ignore_above benefits:
   - Disk space savings
   - Avoiding Lucene term byte-length limits
 - Testing changes:
   - Removed "flaky test" skip annotation
   - Replaced with "continued" test
 - Simplified hits object explanation:
   - Added reference to "Track total hits" docs
 - Query improvements:
   - Added basic multi_match query example before boosting
   - Expanded field boosting explanation in multi_match queries
 - Other clarifications:
   - More concise .keyword suffix usage tip
   - Added term query case sensitivity note

leemthompo commented:
@elasticmachine test this

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

kderusso reviewed (approved):
LGTM, thanks for all the iteration!

shainaraskas commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,611 @@
+          "rating": 4.8
+        }
+      }
+    ]
```
```suggestion
    ]
  }
}
```

shainaraskas committed:
fix yaml

shainaraskas commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,613 @@
+Note that `text` fields are analyzed and multi-fields are generated at index time.
+
+[source,console]
+----
```
it says it's this test that's failing but it's running fine in my environment

leemthompo replied:
😾

leemthompo replied:
[Related subject](https://github.com/user-attachments/assets/80a1e705-40f9-4855-8f08-b20b0505941a)

leemthompo committed:
Fix docscheck?

leemthompo committed:
Is explicit analyzer here messing things up?

leemthompo committed:
Fix wording

elasticmachine committed:
Merge branch 'main' into qs-full-text

leemthompo committed:
Desperado

leemthompo committed:
Change query

leemthompo committed:
Split index creation + mapping steps and add refresh param for better initialization

leemthompo committed:
Update result text

leemthompo committed:
Fix took time

Mikep86 reviewed (approved):
Great work! I left a few comments about wording and technical edge-cases.

leemthompo committed:
Fix test syntax, json spacing🤞

leemthompo commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,624 @@
+==============
+
+[TIP]
+====
+The `multi_match` query is often recommended over a single `match` query for most text search use cases, as it provides more flexibility and better matches user expectations. It won't work if the multi-field mapping isn't enabled.
+====
```
The query won't work if you don't explicitly map the multi-field or use dynamic mapping, but I've already got a TIP about that somewhere else so I'll remove this confusing line 👍

leemthompo commented on docs/reference/quickstart/full-text-filtering-tutorial.asciidoc:
```diff
@@ -0,0 +1,624 @@
+----
+// TEST[continued]
+<1> The title contains "Spicy", matching one of our `should` conditions and contributing to a higher relevance score.
+<2> The description includes "curry", further increasing the document's relevance.
```
ah cool nice, I'll explain this and add link to best_fields

leemthompo committed:
Fixes per Mike's final pass

leemthompo commented:
For the record @kderusso saved the day by finding that the docstests CI checks were failing due to invalid JSON 🥇

elasticsearchmachine commented:
## 💔 Backport failed
The backport operation could not be completed due to the following error:
```
An unexpected error occurred when attempting to backport this PR.
```

You can use [sqren/backport](https://github.com/sqren/backport) to manually backport by running `backport --upstream elastic/elasticsearch --pr 114353`

leemthompo commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/115738">](https://github.com/elastic/elasticsearch/pull/115738)|
|✅|8.16|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/115739">](https://github.com/elastic/elasticsearch/pull/115739)|
|✅|8.15|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/115740">](https://github.com/elastic/elasticsearch/pull/115740)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.3.0 -->
